### PR TITLE
broodmother or a huge spider

### PR DESCRIPTION
**Pull request by kxvvv**
# PR adds:
- a large spider aka broodmother
- its servants.
- a spider egg. 
- broodmother and spider servants faster in spider web.
- broodmother can spit webs, thus binding the creatures. 

### what's going on in the cocoon?
- the creature looks like a "???"
- the creature can't talk
- speed reduction 0.8
- gives some armor, but vulnerability to fire is increased

# pic:

Screenshot_69

 Screenshot_67

Screenshot_65   Screenshot_68

Screenshot_66

image

# vid:

https://github.com/space-wizards/space-station-14/assets/136233650/f9bd96f0-a67d-4eaf-b2db-5437b42c507c

https://github.com/space-wizards/space-station-14/assets/136233650/010b939f-90d1-4e86-9d83-1a3a6fa66bc1

https://github.com/space-wizards/space-station-14/assets/136233650/5473d7ed-0ed9-4c33-9ec1-9d8affed4077

https://github.com/space-wizards/space-station-14/assets/136233650/d6745e04-e457-43be-8e09-a4b1738ee788

https://github.com/space-wizards/space-station-14/assets/136233650/92ad00c9-4315-42a9-ba53-227a38fdf0af

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: A huge spider queen has been spotted at NT stations!

### CODE REVIEW

**Comment by Aisu9** [comment]
what the hell are those damages, Broodmother deal 47 damage per bite (35 pierce/12 poison) and the servant deal 18 (12 piercing and 8 poison)

**Comment by LankLTE** [comment]
Just for reference, the Space Dragon does 30 damage a hit. I wouldn’t go any higher than that- the Xeno Queen’s 20 may be a good reference point.

**Comment by kxvvv** [comment]
> what the hell are those damages, Broodmother deal 47 damage per bite (35 pierce/12 poison) and the servant deal 18 (12 piercing and 8 poison)

That's what I'm talking about, I need balance tips

**Comment by Nimfar11** [comment]
It's worth correcting the translation at brood servant.

**Comment by GeneralGaws** [comment]
Maybe that spider should be megafauna? But its great pr

**Comment by kxvvv** [comment]
> It's worth correcting the translation at brood servant.

for example?

**Comment by deltanedas** [comment]
holy shit kill it with fire

File: Content.Shared/Movement/Components/SlowContactsComponent.cs [hunk 1]
```diff
@@ -17,6 +17,14 @@ public sealed partial class SlowContactsComponent : Component
...
     public float SprintSpeedModifier = 1.0f;
 
+    [DataField("spiderWalkSpeedModifier"), ViewVariables(VVAccess.ReadWrite)]
+    [AutoNetworkedField]
+    public float SpiderWalkSpeedModifier = 1.5f;
+
+    [AutoNetworkedField]
+    [DataField("spiderSprintSpeedModifier"), ViewVariables(VVAccess.ReadWrite)]
+    public float SpiderSprintSpeedModifier = 1.5f;
+
```
Comment (metalgearsloth) [inline Content.Shared/Movement/Components/SlowContactsComponent.cs:27]: Do not hardcode in specific entities like this as it breaks composition.

If you want it whitelisted then use an entity whitelist.

File: Content.Shared/Movement/Systems/SlowContactsSystem.cs [hunk 2]
```diff
@@ -85,6 +86,14 @@ private void MovementSpeedCheck(EntityUid uid, SlowedByContactComponent componen
             if (!TryComp<SlowContactsComponent>(ent, out var slowContactsComponent))
                 continue;
 
+            if (HasComp<SpiderComponent>(uid) & HasComp<SpiderWebObjectComponent>(ent))
+            {
+                walkSpeed = Math.Max(walkSpeed, slowContactsComponent.SpiderWalkSpeedModifier);
+                sprintSpeed = Math.Max(sprintSpeed, slowContactsComponent.SpiderSprintSpeedModifier);
+                remove = false;
+                continue;
+            }
```
Comment (metalgearsloth) [inline Content.Shared/Movement/Systems/SlowContactsSystem.cs:95]: As above do not hardcode in specific entities.

File: Resources/Prototypes/Entities/Mobs/NPCs/broodmother.yml [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    - map: ["enum.DamageStateVisualLayers.Base"]
+      state: alive
+      sprite: Mobs/Animals/quin.rsi
+    scale: 1.2, 1.2
+  - type: DamageStateVisuals
+    states:
+      Alive:
+        Base: alive
+      Critical:
+        Base: dead
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/NPCs/broodmother.yml:20]: unused, please remove.

File: Resources/Prototypes/Entities/Mobs/NPCs/broodmother.yml [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    mask: /Textures/Effects/LightMasks/cone.png
+    autoRot: true
+  - type: MobThresholds
+    thresholds:
+      0: Alive
+      1000: Dead
+  - type: SlowOnDamage
+    speedModifierThresholds:
+      500: 0.7
+      750: 0.8
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/NPCs/broodmother.yml:38]: Why does it go down then up?

File: Resources/Prototypes/Entities/Mobs/NPCs/broodmother.yml [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    damage:
+      types:
+        Piercing: 35
+        Poison: 12
+  - type: Spider
+    webActionName: BroodWebAction
+  - type: EggLayer
+    hungerUsage: 10
+    eggLayCooldownMin: 9999999
+    eggLayCooldownMax: 99999999
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/NPCs/broodmother.yml:59]: Don't add hacks if you want an infinite cooldown.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by JoeHammad1844** [comment]
god damn that bitch is fast

**Comment by metalgearsloth** [comment]
Is this still draft?

**Comment by kxvvv** [comment]
> Is this still draft?

yes, need fix something in her skill

**Comment by LankLTE** [comment]
@kxvvv Will you be coming back to this?

**Comment by metalgearsloth** [comment]
> > @kxvvv Will you be coming back to this?
> 
> it's done i need review🧐

Your tests are failing.

**Comment by kxvvv** [comment]
too many code updates came out during your "mega very fast" review, I'm too lazy to redo it ._.

**Pull request closed without merging**